mod/state-transition/pkg/core/state_processor.go

```diff
@@ -109,6 +110,9 @@ type StateProcessor[
 	// ones.
 	// We prune the map to preserve only current and previous epoch
 	valSetByEpoch map[math.Epoch][]ValidatorT
+
+	// valSetMu protects valSetByEpoch from concurrent accesses
+	valSetMu sync.RWMutex
```

Comment:
nit: usually a mutex guarding a struct member is put above it 

```suggestion
	// valSetMu protects valSetByEpoch from concurrent accesses
	valSetMu sync.RWMutex
	// We prune the map to preserve only current and previous epoch
	valSetByEpoch map[math.Epoch][]ValidatorT
```

Reply:
Done